lib/Resizable.js

```diff
@@ -4,227 +4,492 @@ import PropTypes from 'prop-types';
+                x, y, canDragX, canDragY, deltaX, deltaY)
+
+            width = dimension.width
+            height = dimension.height
+            top = dimension.top
+            left = dimension.left
+        }
+
+        [width, height] = this.runConstraints(width, height)
+
+        if (!this.hasDimensionChanged(width, height, left, top)) {
+            return;
+        }
+
+        const resizeState = this.onResizeState(width,height,left,top)
+
+        if (!this.onResize) {
+            this.setState(resizeState);
+            return
+        }
+
+        this.persistEventIfFunction(event)
+
+        this.setState(resizeState, () =>
+            this.props.onResize(event,
+                {
+                    node,
+                    size: this.getDimension(resizeState)
+                }));
+
+    }
+
+    onResizeStop = (event, {node}) => {
+        if (!this.props.onResizeStop) {
+            this.setState(this.onResizeStopState());
+            return
+        }
+
+        this.persistEventIfFunction(event)
+
+        this.setState(this.onResizeStopState(), () =>
+            this.props.onResizeStop(event,
+                {
+                    node,
+                    size: this.getDimension(this.state)
+                })
+        );
+    }
+
+    onResizeStart = (event, {node, x, y}) => {
+
+        if (this.isResizeHandle(event.target.className)) {
+            this.setState({resizableHandle: event.target.className})
+        }
+
+        const resizeStartState = this.onResizeStartState(x, y)
+
+        if (!this.props.onResizeStart) {
+            this.setState(resizeStartState);
+            return
+        }
+
+        this.persistEventIfFunction(event)
+
+        this.setState(resizeStartState, () =>
+            this.props.onResizeStart(event,
+                {
+                    node,
+                    size: this.getDimension(this.state)
+                }
+            ));
+    }
+
+    getDimension = (state) => (
+        {width: state.width, height: state.height, left: state.left, top: state.top}
+    )
+
+
+    onResizeState = (width, height, left, top) => (
+        {width, height, left, top}
+    )
+
+    onResizeStopState = () => (
+        {resizableHandle: '', resizing: false, slackW: 0, slackH: 0}
+    )
+
+    onResizeStartState = (startX, startY) => (
+        {resizing: true, startX: startX, startY: startY}
+    )
+
+    persistEventIfFunction = (event) => {
+        if (typeof event.persist === 'function') {
+            event.persist();
+        }
+    }
+
+    hasDimensionChanged = (width, height, left, top) => {
+        return width !== this.state.width ||
+            height !== this.state.height ||
+            left !== this.state.left ||
+            top !== this.state.top
+    }
+
+    isMouseMovingUp = (deltaY ) => deltaY < 0
+    isMouseMovingLeft = (deltaX) => deltaX < 0
+    getComponentPosition = () => [this.state.left, this.state.top]
```

Comment:
It's better no to use array in this case. Object is better

Reply:
done